Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -906,9 +907,22 @@ void GraphicsContextSkia::setLineCap(LineCap lineCap)
     m_skiaState.m_stroke.cap = toSkiaCap(lineCap);
 }
 
+static bool isValidDashArray(const DashArray& dashArray)
+{
+    // See 'dom-context-2d-setlinedash': if the array contains not finite or negative values, return
+    DashArray::value_type total = 0;
+    for (const auto& dash : dashArray) {
+        if (dash < 0 || !std::isfinite(dash))
+            return false;
+        total += dash;
+    }
+    // Nothing to be done for all-zero or empty arrays
+    return total > 0;
```

Comment:
@carlosgcampos , @Scony 

I took the letter of the spec in https://html.spec.whatwg.org/multipage/canvas.html#dom-context-2d-setlinedash, and felt OK about checking for negative and not-finite values in the loop. I then took a look at what Skia does when checking for `total > 0` as that covers empty and all-zero arrays.

While there's some overhead in doing this -- since the Skia internal implementation does it, too, it makes sense, as we're now enforcing what the spec dictates. What do you think?